cmd/dp-dimension-importer/main.go

```diff
@@ -58,28 +81,61 @@ func main() {
 		return repository.NewDimensionRepository(neo4jClient, map[string]string{})
 	}
 
-	importAPI := client.ImportAPI{
-		ResponseBodyReader: responseBodyReader{},
-		HTTPClient:         &http.Client{},
-		AuthToken:          cfg.ImportAuthToken,
-		ImportHost:         cfg.ImportAddr,
-	}
+	// ImportAPI HTTP client.
+	importAPI := client.NewImportAPI(cfg.ImportAddr, cfg.ImportAuthToken, responseBodyReader{}, &http.Client{})
+
+	// Handler for dimensionsExtracted events.
+	eventHandler := handler.NewDimensionExtractedEventHandler(
+		newDimensionInserterFunc,
+		&repository.InstanceRepository{Neo4j: neo4jClient},
+		importAPI)
+
+	// MessageProducer for dimensionsInsertedEvents
+	instanceCompletedProducer := message.NewInstanceCompletedProducer(
+		outgoingInstances,
+		schema.InstanceCompletedSchema,
+	)
 
-	eventHandler := &handler.DimensionsExtractedEventHandler{
-		NewDimensionInserter: newDimensionInserterFunc,
-		InstanceRepository:   &repository.InstanceRepository{Neo4j: neo4jClient},
-		ImportAPI:            importAPI,
+	// Errors handler
+	errorEventHandler := &handler.ErrorHandler{
+		Producer:   errorEventProducer,
+		Marshaller: schema.ErrorEventSchema,
 	}
 
-	dimensionInsertedProducer := message.DimensionInsertedProducer{
-		Producer:   insertedEventProducer,
-		Marshaller: schema.DimensionsInsertedSchema,
+	// HTTP Health check endpoint.
+	healthcheck.NewHandler()
+
+	gracefulShutdown := func() {
+		log.Info(gracefulShutdownMsg, nil)
+		ctx, _ := context.WithTimeout(context.Background(), 5*time.Second)
+
+		message.CloseConsumer()
+		incomingInstances.Close(ctx)
+		outgoingInstances.Close(ctx)
+		errorEventProducer.Close(ctx)
+		healthcheck.Close(ctx)
+
+		log.Info(gracefulShutdownComplete, nil)
+		os.Exit(0)
```

Comment:
We should probably exit with a non-zero exit code to signal something went wrong. I just realised I never done this in the obs importer.